pkg/operator/controllers/internetchecker/internetchecker_controller.go

```diff
@@ -62,64 +71,76 @@ func (r *InternetChecker) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		return reconcile.Result{}, err
 	}
 
-	var condition status.ConditionType
-	switch r.role {
-	case operator.RoleMaster:
-		condition = arov1alpha1.InternetReachableFromMaster
-	case operator.RoleWorker:
-		condition = arov1alpha1.InternetReachableFromWorker
-	}
+	// limit all checks to take no longer than half of the requeueInterval
+	ctx, cancel := context.WithTimeout(context.Background(), requeueInterval/2)
+	defer cancel()
```

Comment:
this is already being done by checkBackoff - suggest using one mechanism to do this rather than two, to make it easier to understand